pysilcam/postprocess.py

```diff
@@ -38,62 +39,107 @@ def d50_from_stats(stats, settings):
-    d50 = d50_from_vd(vd,dias)
+    d50 = d50_from_vd(vd, dias)
     return d50
 
-def d50_from_vd(vd,dias):
+
+def d50_from_vd(vd, dias):
     '''
     Calculate d50 from a volume distribution
-    
+
     Args:
         vd            : particle volume distribution calculated from vd_from_stats()
         dias          : mid-points in the size classes corresponding the the volume distribution,
                                returned from get_size_bins()
-        
+
     Returns:
         d50 (float)                 : the 50th percentile of the cumulative sum of the volume distributon, in microns
     '''
 
     # calculate cumulative sum of the volume distribution
-    csvd = np.cumsum(vd/np.sum(vd))
+    csvd = np.cumsum(vd / np.sum(vd))
 
     # find the 50th percentile and interpolate if necessary
-    d50 = np.interp(0.5,csvd,dias)
+    d50 = np.interp(0.5, csvd, dias)
     return d50
 
+
 def get_size_bins():
     '''
     Retrieve size bins for PSD analysis
-    
+
     Returns:
         bin_mids_um (array)     : mid-points of size bins
         bin_limits_um (array)   : limits of size bins
     '''
     # pre-allocate
-    bin_limits_um = np.zeros((53),dtype=np.float64)
+    bin_limits_um = np.zeros((53), dtype=np.float64)
 
     # define the upper limit of the smallest bin (same as LISST-100x type-c)
     bin_limits_um[0] = 2.72 * 0.91
 
     # loop through 53 size classes and calculate the bin limits
-    for I in np.arange(1,53,1):
+    for i in np.arange(1, 53, 1):
         # each bin is 1.18 * larger than the previous
-        bin_limits_um[I] = bin_limits_um[I-1] * 1.180
+        bin_limits_um[i] = bin_limits_um[i - 1] * 1.180
 
     # pre-allocate
-    bin_mids_um = np.zeros((52),dtype=np.float64)
+    bin_mids_um = np.zeros((52), dtype=np.float64)
 
     # define the middle of the smallest bin (same as LISST-100x type-c)
     bin_mids_um[0] = 2.72
 
     # loop through 53 size classes and calculate the bin mid-points
-    for I in np.arange(1,52,1):
+    for i in np.arange(1, 52, 1):
         # each bin is 1.18 * larger than the previous
-        bin_mids_um[I]=bin_mids_um[I-1]*1.180
+        bin_mids_um[i] = bin_mids_um[i - 1] * 1.180
 
     return bin_mids_um, bin_limits_um
 
-def vd_from_nd(count,psize,sv=1):
+
+def extract_middle(stats, crop_bounds):
+    '''
+    Temporary cropping solution due to small window in AUV
+
+    Args:
+        stats (df)    : silcam stats file
+        crop_bounds (tuple) : 4-tuple of lower-left (row, column) then upper-right (row, column) coord of crop
+
+    Returns:
+        stats (df)    : cropped silcam stats file
+    '''
+
+    # print('initial stats length:', len(stats))
+    r = np.array(((stats['maxr'] - stats['minr']) / 2) + stats['minr'])  # pixel row of middle of bounding box
+    c = np.array(((stats['maxc'] - stats['minc']) / 2) + stats['minc'])  # pixel column of middle of bounding box
+
+    points = []
+    for i in range(len(c)):
+        points.append([(r[i], c[i])])
```

Comment:
Perhaps a more concise list comprehension solution: points = [[(r_, c_)] for r_, c_ in zip(r, c)] ?

Reply:
Or, since an array is the final datastructure (below): pts = np.vstack([r, c])

Reply:
I did the first one here. The second one was confusing to me, so I thought I'd be lazy and pretend we keep the easy option for code readability.